### PR TITLE
Custom font selection

### PR DESCRIPTION
**Pull request by JorikSchellekens**
Screen Recording 2020-06-15 at 15 32 45

### CODE REVIEW

**Comment by JorikSchellekens** [comment]
image

**Comment by ara4n** [comment]
Does this let you specify a series of fonts (for fallback purposes - e.g. for folks who want to default to Nunito but have a different font for CJK langs or similar?)

**Comment by JorikSchellekens** [comment]
@ara4n completely overrides font family so that's entirely possible by specifying a comma separated list of fonts. However it doesn't do any checks so user foot guns abound

File: src/components/views/elements/Field.tsx [hunk 1]
```diff
@@ -242,10 +244,10 @@ export default class Field extends React.PureComponent<PropShapes, IState> {
         const Tooltip = sdk.getComponent("elements.Tooltip");
         let fieldTooltip;
         if (tooltipContent || this.state.feedback) {
-            const addlClassName = tooltipClassName ? tooltipClassName : '';
+            const addClassName = tooltipClassName ? tooltipClassName : '';
```
Comment (jryans) [inline src/components/views/elements/Field.tsx:247]: I think `addl` here meant additional. Maybe let's avoid the naming question entirely and use the `classNames` function we use elsewhere for this...?

File: src/dispatcher/actions.ts [hunk 2]
```diff
@@ -69,4 +69,14 @@ export enum Action {
      * Opens the user menu (previously known as the top left menu). No additional payload information required.
      */
     ToggleUserMenu = "toggle_user_menu",
+
+    /**
+     * Sets the apps root font size. Should be used with UpdateFontSizePayload
+     */
+    UpdateFontSize = "update_font_size",
```
Comment (jryans) [inline src/dispatcher/actions.ts:76]: Should this font size part be in a different PR...?

File: src/components/views/elements/Field.tsx (hunk 1 above)
Comment (JorikSchellekens) [reply src/components/views/elements/Field.tsx:247]: Oh, I thought it must have been a typo I introduced. Sure, let's use `classnames`

File: src/dispatcher/actions.ts (hunk 2 above)
Comment (JorikSchellekens) [reply src/dispatcher/actions.ts:76]: This doesn't change the behaviour of font sizing in any way. I just thought I would opportunistically move this to type-checked ts while I was in the area.
